### PR TITLE
Add codeowners for `Lib/ensurepip/`

### PR DESCRIPTION
**Pull request by AlexWaygood**
There was some recent concern on the core dev Discord server about PRs from third-party contributors attempting to update the bundled version of pip. While sometimes it _might_ be okay to accept these PRs, it's good for the pip maintainers to be kept in the loop whenever a PR like that is made from a third-party contributor. As such, this PR adds as `ensurepip` codeowners the three pip maintainers who are also core devs: @pradyunsg, @pfmoore and @dstufft. This means that all three will be automatically pinged for review whenever anybody creates a PR that touches files in the `Lib/ensurepip/` directory.

@pradyunsg, @pfmoore, @dstufft: please confirm you're okay with being added as codeowners! I'll obviously remove anybody who doesn't want to be added here :)

### CODE REVIEW

**Review by pfmoore — APPROVED** [review]

**Review by ezio-melotti — APPROVED** [review]

**Review by schribl — APPROVED** [review]
Even so my review does not count, I like this! As it also will make it easier for external contributors to have a clear checkmark that the contribution is good.

As I felt I made a possible mistake to create a PR for a third-party vendored library.

**Review by pradyunsg — APPROVED** [review]

**Comment by pradyunsg** [comment]
> There was some recent concern on the core dev Discord server about PRs from third-party contributors attempting to update the bundled version of pip.

FWIW, this concern was somewhat unnecessary. 😅

We've got CI checks to ensure that the bundled copy of pip matches the one on PyPI: https://github.com/python/cpython/issues/112721#issuecomment-1839347787

**Comment by AlexWaygood** [comment]
:wave: friendly ping @dstufft -- are you okay with being added as a codeowner? Happy to remove you if not :)

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -157,6 +157,8 @@ Doc/c-api/stable.rst          @encukou
 
 **/*dataclasses*              @ericvsmith
 
+**/*ensurepip*                @pfmoore @pradyunsg @dstufft
```
Comment (AlexWaygood) [inline .github/CODEOWNERS]:
```suggestion
**/*ensurepip*                @pfmoore @pradyunsg
```

**Comment by AlexWaygood** [comment]
Donald hasn't responded regarding whether or not he's okay with being added as a codeowner, so I've removed his username for now. It can always be added later :)

**Pull request merged**